app/my-dashboard/_features/rewards-table/rewards-table.tsx

```diff
@@ -0,0 +1,114 @@
+import { getCoreRowModel, useReactTable } from "@tanstack/react-table";
+import { useMemo, useState } from "react";
+
+import { RewardReactQueryAdapter } from "@/core/application/react-query-adapter/reward";
+import { GetProjectRewardsPortParams, GetRewardsQueryParams } from "@/core/domain/reward/reward-contract.types";
+
+import { Typo } from "@/design-system/atoms/typo";
+import { Table, TableLoading } from "@/design-system/molecules/table";
+import { TableSearch } from "@/design-system/molecules/table-search";
+
+import { ErrorState } from "@/shared/components/error-state/error-state";
+import { ScrollView } from "@/shared/components/scroll-view/scroll-view";
+import { ShowMore } from "@/shared/components/show-more/show-more";
+import { FilterButton } from "@/shared/features/filters/_components/filter-button/filter-button";
+import { FilterDataProvider } from "@/shared/features/filters/_contexts/filter-data/filter-data.context";
+import { useAuthUser } from "@/shared/hooks/auth/use-auth-user";
+
+import { FilterColumns } from "./_components/filter-columns/filter-columns";
+import { useFilterColumns } from "./_components/filter-columns/filter-columns.hooks";
+import { FilterData } from "./_components/filter-data/filter-data";
+import { useProjectRewardsFilterDataSidePanel } from "./_components/filter-data/filter-data.hooks";
+
+export type RewardsTableFilters = Omit<
+  NonNullable<GetProjectRewardsPortParams["queryParams"]>,
+  "pageSize" | "pageIndex"
+>;
+
+// TODO Update table @sami
+export function RewardsTable() {
+  const [search, setSearch] = useState<string>();
+  const [debouncedSearch, setDebouncedSearch] = useState<string>();
+  const { open: openFilterPanel } = useProjectRewardsFilterDataSidePanel();
+  // const { open } = useRewardDetailSidepanel();
+  const [filters, setFilters] = useState<RewardsTableFilters>({});
+  const { githubUserId } = useAuthUser();
+
+  const { columns, selectedIds, setSelectedIds, sorting, setSorting, sortingParams } = useFilterColumns();
+
+  const queryParams: Partial<GetRewardsQueryParams> = {
+    search: debouncedSearch,
+    ...filters,
+    ...sortingParams,
+    recipientIds: [githubUserId ?? 0],
```

Comment:
```suggestion
    recipientIds: githubUserId ? [githubUserId] : undefined,
```

Wouldn't this be safer ?